File: dotnet/cafdemalihapa/Directories/Hosting/Feature/Data/DataDirectory.cs
namespace cafdemalihapa.Directories.Hosting.Feature.Data
{
    public class DataDirectory()
    {
        public string GetName()
        {
            return "data";
        }
        public string GetPath()
        {
            var directoryName = GetName();
            var featureDirectoryPath = FeatureDirectory.GetPath();
            var dataDirectory = Path.Combine(featureDirectoryPath, directoryName);
            return dataDirectory;
        }
        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/Data/DataDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.Data

open System.IO
open cafdemwimu.console.Directories.Hosting.Feature

type DataDirectory() =
    member _.GetName() = "data"

    member this.GetPath() =
        let directoryName = this.GetName()
        let featureDirectoryPath = FeatureDirectory.GetPath()
        let dataDirectory = Path.Combine(featureDirectoryPath, directoryName)
        dataDirectory

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore
